core-foundation/src/dictionary.rs

```diff
@@ -63,50 +84,30 @@ impl<K, V> CFDictionary<K, V> {
     }
 
     #[inline]
-    pub fn contains_key(&self, key: *const c_void) -> bool {
-        unsafe { CFDictionaryContainsKey(self.0, key) != 0 }
-    }
-
-    /// Similar to `contains_key` but acts on a higher level, automatically converting from any
-    /// `TCFType` to the raw pointer of its concrete TypeRef.
-    #[inline]
-    pub fn contains_key2(&self, key: &K) -> bool where K: TCFType {
-        self.contains_key(key.as_concrete_TypeRef().as_void_ptr())
+    pub fn contains_key(&self, key: &K) -> bool where K: ToVoid {
+        unsafe { CFDictionaryContainsKey(self.0, key.to_void()) != 0 }
     }
 
     #[inline]
-    pub fn find(&self, key: *const c_void) -> Option<*const c_void> {
+    pub fn find<'a>(&self, key: &K) -> Option<ItemRef<'a, V>> where V: FromVoid, K: ToVoid {
```

Comment:
Seeing no use of `'a` in the input arguments is surprising. Should it be related to `&self`?